openmdao/docs/openmdao_book/features/building_blocks/components/subproblem_comp.ipynb

```diff
@@ -0,0 +1,235 @@
+{
```

Comment:
<p><code>SubproblemComp</code> provides a way to evaluate an OpenMDAO system within an OpenMDAO system. Unlike a typical sub-system, using a subproblem exposes only a limited number of inputs and outputs of the underlying model to the parent system. This may be beneficial from a performance standpoint when there are many inputs and outputs of the internal system that aren't needed in the external system.</p><p> </p><p>Note that this means that nonlinear and linear solvers are needed on the subproblem's model if implicit behavior is present, since the internal variables are invisible to solvers in the parent system.</p><p> </p><p>At this time, OpenMDAO does not evaluate gradients through the optimization process, so a gradient-based driver at the outer level cannot access derivatives across the optimization on the inner level.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OpenMDAO/OpenMDAO/pull/2817/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Reply:
Added this under 'Using SubproblemComp' section